### ISSUE TITLE
Missing `has()` check in `keys()`

### ISSUE THREAD

**Issue by nemoDreamer** [jasmine/jasmine#1194]
Hey there,
`.toEqual()` is failing for me since updating to edge, and I tracked it down to a missing `has()` in the `keys` function.

**Note:** I tried to fork and contribute, but `jasmine-core` package seems to itself load a (gitignored) older version of itself inside of its own `./node_modules`, so editing `./lib/jasmine-core/jasmine.js` has no effect, and a change in `/node_modules/jasmine-core//node_modules/jasmine-core/lib/jasmine-core/jsmine.js` can't of course get committed...

Further, I found that the non-node_modules version of that lib has already been fixed using a `for` loop instead of `for-in`, but that newer file doesn't get loaded since the node-module dependency takes priority for some reason.

Previous version (commented out hotfix I added):

```javascript
var extraKeys = [];
for (var i in allKeys) {
    // if (has(allKeys, i)) {
        if (!allKeys[i].match(/^[0-9]+$/)) {
            extraKeys.push(allKeys[i]);
        }
    // }
}
```

Seems to have been changed recently, but doesn't get loaded:

```javascript
var extraKeys = [];
if (allKeys.length === 0) {
    return allKeys;
}

for (var x = 0; x < allKeys.length; x++) {
    if (!allKeys[x].match(/^[0-9]+$/)) {
        extraKeys.push(allKeys[x]);
    }
}
```

**Comment by mbuckley** [comment]
Hi @nemoDreamer I'm seeing this issue and just opened a ticket for it here. https://github.com/jasmine/jasmine/issues/1195. We're seeing this when updating from v2.4.1 to v2.5.0.

**Comment by nemoDreamer** [comment]
I guess a more general question is: why is the `jasmine-core` node-module including an older version of `jasmine-core` inside of itself, and why is that older version's `/lib` being loaded by Karma, instead of the `/lib` in its root? 😢

**Comment by slackersoft** [comment]
Closing this as a duplicate of #1188. The fix should be released in v2.5.1

**Comment by nemoDreamer** [comment]
Thanks, guys, I just updated to v2.5.1, and all's ✨ !
